src/main/java/tutoraid/logic/commands/EditLessonCommand.java

```diff
@@ -82,17 +76,13 @@ public CommandResult execute(Model model) throws CommandException {
 
         Capacity newCapacity = editedLesson.getCapacity();
         Students currStudents = lessonToEdit.getStudents();
-
         if (newCapacity.getCapacity() < currStudents.numberOfStudents()) {
             throw new CommandException(MESSAGE_CAPACITY_LESS_THAN_STUDENTS);
         }
-
-        model.setLesson(lessonToEdit, editedLesson);
-        Student.updateStudentLessonLink(studentList, lessonToEdit, editedLesson);
-        model.viewLesson(editedLesson);
-        model.updateFilteredStudentList(editedLesson::hasStudent);
-
-        return new CommandResult(String.format(MESSAGE_EDIT_LESSON_SUCCESS, editedLesson.toNameString()));
+        lessonToEdit.replace(editedLesson);
```

Comment:
Update the fields of the object directly, as this object is linked to the Student